holoviews/core/data/xarray.py

```diff
@@ -172,7 +173,17 @@ def retrieve_unit_and_label(dim):
                 for vdim in vdims:
                     arr = data[vdim.name]
                     if not isinstance(arr, xr.DataArray):
-                        arr = xr.DataArray(arr, coords=coords, **xr_kwargs)
+                        with warnings.catch_warnings():
+                            # xarray emit a warning when datetime/timedelta is not nanosecond,
+                            # and then convert it to nanosecond precision. The full warning is:
+                            #   Converting non-nanosecond precision {case} values to nanosecond precision.
+                            #   This behavior can eventually be relaxed in xarray, as it is an artifact from
+                            #   pandas which is now beginning to support non-nanosecond precision values.
+                            #   This warning is caused by passing non-nanosecond np.datetime64 or
+                            #   np.timedelta64 values to the DataArray or Variable constructor; it can be
+                            #   silenced by converting the values to nanosecond precision ahead of time.
+                            warnings.filterwarnings("ignore", "Converting non-nanosecond precision")
+                            arr = xr.DataArray(arr, coords=coords, **xr_kwargs)
```

Comment:
I don't know if this is the correct way to do it. Alternative ways:

1) Ignore the warning in pyproject.toml
2) Convert the data in the unit test to nanoseconds beforehand.